### ISSUE TITLE
tag redundant , does notnull need use_zero to handle zero or false?

### ISSUE THREAD

**Issue by focusonline** [go-pg/pg#1507]
vmihailenco,
I am confused about notnull and use_zero,  when I used version 8 of go-pg, it worked fine without use_zero.
actually if a field has "not null" defined there must be a value stored in it, no matter it is zero or false or otherelse, any value shoule be taken and updated except NULL. regardless what the type of field is, boolean string or integer, notnull should take all value except NULL
what is the use_zero for? why introduced this tag? seems not null is enough to describe the requirement.
anyway, if upgrading from 8 to 9, there are something changed without comparision and description. if the discrepancy presented, will be big help and appreciated.

**Comment by vmihailenco** [comment]
Please read information that is provided in changelog and let me know what is unclear and should be explained better.

**Comment by focusonline** [comment]
Dear vmihailenco
thank you for reply quickly, it is only an advice. I modified the code today, remove all sql: blahblah tags for pg: , so the bugs disappeared. this time, I upgraded from v8 to v9, somethings changed introduced some bugs, a little bit bumped not so smoothly.
As I mentioned, use_zero is one of the bug, in V8, it can insert 0 into notnull fields perfectly, but in v9 without use_zero, it always inserts NULL into same fields (althought there is notnull tag indicated). Just a little confused for the necessity and compatibility.
I noticed there are some warning messages printed by go-pg like XXX deprecated, it is very helpful to understand what is changed.
However, go-pg is awesome, it is very smart and convenient,  I like it so much, and thank you for your effort again.
